assets/shared/blocks/settings.js

```diff
@@ -86,3 +90,59 @@ export const withDefaultBlockStyle = () => ( Component ) => ( props ) => {
 
 	return <Component { ...props } { ...extraProps } />;
 };
+
+/**
+ * This HOC sets the default color attribute to a value based in a probe.
+ *
+ * @param {Object} colorConfigs Colors config object, where the key is the
+ *                              default color attribute name, and the value is
+ *                              an object containing style and probeKey.
+ *                              The block attributes must register an attribute
+ *                              for every key.
+ *
+ * @return {Function} Extended component.
+ */
+export const withDefaultColor = ( colorConfigs ) => ( Component ) => (
+	props
+) => {
+	const { setAttributes, attributes } = props;
+	const colorSlugsByProbe = useColorSlugsByProbe();
+	const [ colorProps, setColorProps ] = useState( {} );
+
+	const colorConfigsDeps = Object.keys( colorConfigs ).map(
+		( key ) => attributes[ key ]
+	);
+
+	useEffect( () => {
+		const newColorProps = {};
+
+		Object.entries( colorConfigs ).forEach(
+			( [ colorKey, { style, probeKey } ] ) => {
+				const probeColorSlug = colorSlugsByProbe[ probeKey ];
+
+				newColorProps[ colorKey ] = {
+					slug: attributes[ colorKey ],
+					className: getColorClassName(
```

Comment:
I tried to use `class` here to follow the `withColors` standard, but it was causing some eslint errors while using, so I decided to use `className`.